lib/cmd_tool_next.go

```diff
@@ -0,0 +1,79 @@
+package lib
+
+import (
+	"fmt"
+	"net"
+	"github.com/spf13/pflag"
+)
+
+type CmdToolNextFlags struct {
+	Help  bool
+	Quiet bool
+}
+
+func (f *CmdToolNextFlags) Init() {
+	pflag.BoolVarP(
+		&f.Help,
+		"help", "h", false,
+		"show help.",
+	)
+	pflag.BoolVarP(
+		&f.Quiet,
+		"quiet", "q", false,
+		"quiet mode; suppress additional output.",
+	)
+}
+
+func CmdToolNext(
+	f CmdToolNextFlags,
+	args []string,
+	printHelp func(),
+) error {
+	if f.Help {
+		printHelp()
+		return nil
+	}
+
+	increment := 1
+	actionFunc := func(input string, inputType INPUT_TYPE) error {
+		switch inputType {
+		case INPUT_TYPE_IP:
+			UpdateIPAddress(input, increment)
+		default:
+			return ErrNotIP
+		}
+		return nil
+	}
+	err := GetInputFrom(args, true, true, actionFunc)
+	if err != nil {
+		fmt.Println(err)
+	}
+	return nil
+}
+
+func UpdateIPAddress(input string, delta int) {
```

Comment:
This algorithm is confusing me because I think it should be quite simple: convert the IP into the 32-bit (integer) format, and do the plus/minus. It can overflow as well, to that case must be taken care of (max IP's "next" is 0, and 0 IP's "previous" is the max IP).

Also this must support v6.